packages/protocol/contracts/common/UniswapFeeHandlerSeller.sol

```diff
@@ -0,0 +1,158 @@
+pragma solidity ^0.5.13;
+
+import "../common/interfaces/IFeeHandlerSeller.sol";
+
+import "../stability/interfaces/IExchange.sol";
+import "../stability/interfaces/ISortedOracles.sol";
+import "./UsingRegistry.sol";
+import "../stability/StableToken.sol";
+import "../common/FixidityLib.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+import "../common/Initializable.sol";
+
+import "../uniswap/interfaces/IUniswapV2RouterMin.sol";
+import "../uniswap/interfaces/IUniswapV2FactoryMin.sol";
+
+contract UniswapFeeHandlerSeller is IFeeHandlerSeller, UsingRegistry, Initializable {
+  using FixidityLib for FixidityLib.Fraction;
+  mapping(address => address[]) public routerAddresses;
+
+  uint256 constant MAX_TIMESTAMP_BLOCK_EXCHANGE = 20;
+
+  event ReceivedQuote(address router, uint256 quote);
+  event RouterUsed(address router);
+
+  /**
+   * @notice Sets initialized == true on implementation contracts.
+   * @param test Set to true to skip implementation initialisation.
+   */
+  constructor(bool test) public Initializable(test) {}
+
+  function initialize(address _registryAddress) external initializer {
+    _transferOwnership(msg.sender);
+    setRegistry(_registryAddress);
+  }
+
+  // This function explicitly defines few variables because it was getting error "stack too deep"
+  function sell(
+    address sellTokenAddress,
+    address buyToken,
+    uint256 amount,
+    uint256 maxSlippage // as fraction,
+  ) external {
+    require(
+      buyToken == registry.getAddressForOrDie(GOLD_TOKEN_REGISTRY_ID),
+      "Buy token can only be gold token"
+    );
+
+    // An improvement to this function would be to allow the user to pass a path as argument
+    // and if it generates a better outcome that the ones enabled that gets used
+    // and the user gets a reward
+
+    IERC20 goldToken = getGoldToken();
+    // address celoAddress = address(getGoldToken());
+
+    uint256 bestRouterIndex = 0;
+    uint256 bestRouterQuote = 0;
+
+    address[] memory path = new address[](2);
+    // address[] memory thisTokenRouterAddresses = routerAddresses[sellTokenAddress];
+
+    require(routerAddresses[sellTokenAddress].length > 0, "routerAddresses should be non empty");
```

Comment:
We could move this require to the top to save a bit of gas if routerAddresses are not present